### ISSUE TITLE
Review insecure uses of .interpret in class library

### ISSUE THREAD

**Issue by snappizz** [supercollider/supercollider#2869]
There are a number of instances where `.interpret` is used inappropriately in the core lib, for example in `Color:fromHexString`.

**Comment by brianlheim** [comment]
Can we limit this to a few specific, enumerated cases? And provide a definition of "insecure use"?

```supercollider
Class.findAllReferences(\interpret).do(_.postln)
```

```
Document:prAdd // valid use
EZControlSpecEditor:init // should probably be an integer parse?
EZText:init // not sure, looks like it should be an integer parse
Function:makeFlopFunc // looks like it could be done more simply
HistoryGui:makeViews
Meta_Color:fromHexString
Meta_FileReader:readInterpret
Meta_HIDUsage:readHUTFile
Meta_History:unformatTime
NdefGui:makeNameView
NdefParamGui:dragAction
```

**Comment by snappizz** [comment]
Nah, I think this issue is fine as is. Like the RTAlloc and Ctor issues, it's a bunch of small related issues grouped in one

**Comment by brianlheim** [comment]
The scope and motivation are unclear, that's all I'm saying. If you plan to work on this then nvm.

**Comment by snappizz** [comment]
It's not a particularly focused issue -- it's just that `String:interpret` is often a sign of bad code and, more importantly, sources of remote code execution vulnerabilities.

**Comment by telephon** [comment]
Well, in Lisp, you have `eval` all over the place. But good to inspect the cases.

Some might be just an `asFloat` (e.g. Meta_History:unformatTime).

>Function:makeFlopFunc // looks like it could be done more simply

I'd be interested how you would simplify this: make sure that the argument names are there, they are needed.

**Comment by vivid-synth** [comment]
>  it's just that String:interpret is often a sign of bad code and, more importantly, sources of remote code execution vulnerabilities.

Side note: until we have TCP as the default, basically all SC is wide open to remote code execution through sending UDP to the server (e.g. write malicious code with `/b_write` and get the server to call it with a buffer overflow or other out-of-bounds code the server doesn't check for (or something with a UGen function pointer?))

**Comment by brianlheim** [comment]
After #3527, the new list of methods using interpret is:

```
Document:prAdd
EZControlSpecEditor:init
EZText:init
Function:makeFlopFunc
HistoryGui:makeViews
Meta_FileReader:readInterpret
Meta_HIDUsage:readHUTFile
NdefGui:makeNameView
NdefParamGui:dragAction
```
